homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,216 @@
+        }
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._group_number
+
+    @property
+    def min_temp(self):
+        """Return Minimum Temperature for AC of this group."""
+        return self._airtouch.acs[self._unit.BelongsToAc].MinSetpoint
+
+    @property
+    def max_temp(self):
+        """Return Max Temperature for AC of this group."""
+        return self._airtouch.acs[self._unit.BelongsToAc].MaxSetpoint
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._unit.GroupName
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._unit.Temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we are trying to reach."""
+        return self._unit.TargetSetpoint
+
+    @property
+    def hvac_mode(self):
+        """Return hvac target hvac state."""
+        is_on = self._unit.PowerState == "On"
+        if not is_on:
+            return HVAC_MODE_OFF
+
+        return AT_TO_HA_STATE[self._airtouch.acs[self._unit.BelongsToAc].AcMode]
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        airtouch_modes = self._airtouch.GetSupportedCoolingModesByGroup(
+            self._group_number
+        )
+        modes = [AT_TO_HA_STATE[mode] for mode in airtouch_modes]
+        modes.extend([HVAC_MODE_OFF])
+        return modes
+
+    @property
+    def fan_mode(self):
+        """Return fan mode of the AC this group belongs to."""
+        return AT_TO_HA_FAN_SPEED[self._airtouch.acs[self._unit.BelongsToAc].AcFanSpeed]
+
+    @property
+    def fan_modes(self):
+        """Return the list of available fan modes."""
+        airtouch_fan_speeds = self._airtouch.GetSupportedFanSpeedsByGroup(
+            self._group_number
+        )
+        return [AT_TO_HA_FAN_SPEED[speed] for speed in airtouch_fan_speeds]
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            _LOGGER.debug("Setting temp of %s to %s", self._group_number, str(temp))
+            self._unit = self._airtouch.SetGroupToTemperature(
+                self._group_number, int(temp)
+            )
+            self.async_write_ha_state()
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new fan mode."""
+        _LOGGER.debug("Setting fan mode of %s to %s", self._group_number, fan_mode)
+        self._unit = self._airtouch.SetFanSpeedByGroup(
```

Comment:
Consider throwing ValueError if fan_mode is not in the list of supported FAN modes. I think this is possible by direct service calls (to pass bad values).

Reply:
This did cause exceptions when I tested it, have added protection to make sure the fan mode is in the fan modes supported by the air conditioner, and also in the fan modes supported in the dictionary